Wikipedia/Code/WMFWelcomeLanguageViewController.m

```diff
@@ -22,38 +22,47 @@ - (void)viewDidLoad {
     [super viewDidLoad];
     self.languageTableView.editing = YES;
 
-    self.titleLabel.text = MWLocalizedString(@"welcome-languages-title", nil);
-    [self.moreLanguagesButton setTitle:MWLocalizedString(@"welcome-languages-more-languages-button-title", nil) forState:UIControlStateNormal];
-    [self.nextStepButton setTitle:MWLocalizedString(@"welcome-languages-button-title", nil) forState:UIControlStateNormal];
-    self.footnoteLabel.text = MWLocalizedString(@"welcome-languages-footnote-text", nil);
-    [self.howThisWorksButton setTitle:MWLocalizedString(@"welcome-languages-more-info-button-text", nil) forState:UIControlStateNormal];
-}
+    self.titleLabel.text =
+        [MWLocalizedString(@"welcome-languages-title", nil) uppercaseStringWithLocale:[NSLocale currentLocale]];
 
-- (void)viewWillDisappear:(BOOL)animated {
-    [super viewWillDisappear:animated];
-    if ([[MWKLanguageLinkController sharedInstance].preferredLanguages count] > 1) {
-        [[NSUserDefaults standardUserDefaults] wmf_setShowSearchLanguageBar:YES];
-    } else {
-        [[NSUserDefaults standardUserDefaults] wmf_setShowSearchLanguageBar:NO];
-    }
+    self.subTitleLabel.text = MWLocalizedString(@"welcome-languages-sub-title", nil);
+
+    [self.moreLanguagesButton setTitle:MWLocalizedString(@"welcome-languages-add-button", nil)
+                              forState:UIControlStateNormal];
+    
+    [self.nextStepButton setTitle:[MWLocalizedString(@"welcome-languages-continue-button", nil) uppercaseStringWithLocale:[NSLocale currentLocale]]
+                         forState:UIControlStateNormal];
+    
+    self.nextStepButton.backgroundColor = [UIColor wmf_colorWithHex:0xE8F3FE alpha:1.0];
```

Comment:
something to keep in mind: all the "next step" buttons share some common elements. maybe once this starts to solidify, you can go back and refactor the common elements into color/font style categories and maybe a `UIButton` factory method.